src/main/java/org/gridsuite/dynamicsecurityanalysis/server/service/ParametersService.java

```diff
@@ -0,0 +1,203 @@
+import org.gridsuite.dynamicsecurityanalysis.server.entities.parameters.DynamicSecurityAnalysisParametersEntity;
+import org.gridsuite.dynamicsecurityanalysis.server.repositories.DynamicSecurityAnalysisParametersRepository;
+import org.gridsuite.dynamicsecurityanalysis.server.service.contexts.DynamicSecurityAnalysisRunContext;
+import org.gridsuite.dynamicsecurityanalysis.server.utils.Utils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.gridsuite.dynamicsecurityanalysis.server.DynamicSecurityAnalysisException.Type.*;
+
+/**
+ * @author Thang PHAM <quyet-thang.pham at rte-france.com>
+ */
+@Service
+public class ParametersService {
+
+    public static final String MSG_PARAMETERS_UUID_NOT_FOUND = "Parameters uuid not found: ";
+
+    private final String defaultProvider;
+
+    private final DynamicSecurityAnalysisParametersRepository dynamicSecurityAnalysisParametersRepository;
+
+    @Autowired
+    public ParametersService(@Value("${dynamic-security-analysis.default-provider}") String defaultProvider, DynamicSecurityAnalysisParametersRepository dynamicSecurityAnalysisParametersRepository) {
+        this.defaultProvider = defaultProvider;
+        this.dynamicSecurityAnalysisParametersRepository = dynamicSecurityAnalysisParametersRepository;
+    }
+
+    public DynamicSecurityAnalysisRunContext createRunContext(UUID networkUuid, String variantId, String receiver,
+                                                              String provider, ReportInfos reportInfos, String userId,
+                                                              UUID dynamicSimulationResultUuid,
+                                                              UUID dynamicSecurityAnalysisParametersUuid) {
+
+        // get parameters from the local database
+        DynamicSecurityAnalysisParametersInfos dynamicSecurityAnalysisParametersInfos = getParameters(dynamicSecurityAnalysisParametersUuid);
+
+        // build run context
+        DynamicSecurityAnalysisRunContext runContext = DynamicSecurityAnalysisRunContext.builder()
+                .networkUuid(networkUuid)
+                .variantId(variantId)
+                .receiver(receiver)
+                .dynamicSimulationResultUuid(dynamicSimulationResultUuid)
+                .reportInfos(reportInfos)
+                .userId(userId)
+                .parameters(dynamicSecurityAnalysisParametersInfos)
+                .build();
+
+        // set provider for run context
+        String providerToUse = provider;
+        if (providerToUse == null) {
+            providerToUse = runContext.getParameters().getProvider();
+        }
+        if (providerToUse == null) {
```

Comment:
why don't you concatenate this if with the one below you can do something like : 

```
        if (providerToUse == null) {
            providerToUse = runContext.getParameters().getProvider() != null ? runContext.getParameters().getProvider() : defaultProvider;
        }
```
or you can even use another approche : 

```
providerToUse = Optional.ofNullable(runContext)
        .map(RunContext::getParameters)
        .map(Parameters::getProvider)
        .orElse(defaultProvider);
```

Reply:
```suggestion
        if (providerToUse == null) {
            providerToUse = Optional.ofNullable(runContext.getParameters().getProvider()).orElse(defaultProvider);
        }
```
context, parameter always not null before this line, so I propose as above